src/components/Footer.tsx

```diff
@@ -0,0 +1,56 @@
+import { useEffect, useState } from 'react';
+import { filters } from '../constants/filters';
+import { useTodos } from '../hooks/useTodos';
+import { filterTodos } from '../helpers/filterTodos';
+import { FilterBy } from '../types/FilterBy';
+
+export const Footer = () => {
+  const { filterBy, setFilterBy, todos, setTodos, activeTodosAmount } =
+    useTodos();
+  const [isButtonDisabled, setIsButtonDisabled] = useState(false);
+
+  useEffect(() => {
+    const hasCompletedTodos = todos.filter(t => t.completed).length !== 0;
+
+    setIsButtonDisabled(!hasCompletedTodos);
+  }, [todos]);
```

Comment:
```suggestion
    const isButtonDisabled = !todos.filter(t => t.completed).length;
```

redundant state. enought to check completed todos length;